[ObjCRuntime] Unwrap TargetInvocationExceptions we get when invoking methods using reflection.

rolfbjarne opened the pull request:
Fixes this unit test:

    [FAIL] ManagedExceptionPassthrough :   exception
     Expected: same as <System.ApplicationException: 3,14
      at MonoTouchFixtures.ObjCRuntime.ExceptionsTest.ManagedExceptionTest.ThrowManagedException() in xamarin-macios/tests/monotouch-test/ObjCRuntime/ExceptionsTest.cs:line 128>
     But was:  <System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
    ---> System.ApplicationException: 3,14
      at MonoTouchFixtures.ObjCRuntime.ExceptionsTest.ManagedExceptionTest.ThrowManagedException() in xamarin-macios/tests/monotouch-test/ObjCRuntime/ExceptionsTest.cs:line 128
      --- End of inner exception stack trace ---
      at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
      at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture) in System.Private.CoreLib.dll:token 0x6004d2a+0x0
      at ObjCRuntime.Runtime.InvokeMethod(MethodBase method, Object instance, IntPtr native_parameters) in Xamarin.Mac.dll:token 0x60011d1+0x254
      at ObjCRuntime.Runtime.InvokeMethod(MonoObject* methodobj, MonoObject* instanceobj, IntPtr native_parameters) in Xamarin.Mac.dll:token 0x60011d0+0x0
      at ObjCRuntime.Runtime.bridge_runtime_invoke_method(MonoObject* method, MonoObject* instance, IntPtr parameters, IntPtr& exception_gchandle) in Xamarin.Mac.dll:token 0x6001124+0x26
      at ApiDefinition.Messaging.void_objc_msgSendSuper(IntPtr receiver, IntPtr selector)
      at Bindings.Test.ObjCExceptionTest.InvokeManagedExceptionThrower() in xamarin-macios/tests/xharness/tmp-test-dir/monotouch-test2683/obj/Debug/net6.0-macos/macOS/Bindings.Test/ObjCExceptionTest.g.cs:line 105
      at MonoTouchFixtures.ObjCRuntime.ExceptionsTest.ManagedExceptionPassthrough() in xamarin-macios/tests/monotouch-test/ObjCRuntime/ExceptionsTest.cs:line 154>
      at MonoTouchFixtures.ObjCRuntime.ExceptionsTest.ManagedExceptionPassthrough() in xamarin-macios/tests/monotouch-test/ObjCRuntime/ExceptionsTest.cs:line 159

vs-mobiletools-engineering-service2 commented:
### :x: [PR Build] Tests failed on Build :x:

Tests failed on Build. 
* [Azure DevOps](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4812980&view=ms.vss-test-web.test-result-details)
* [Html Report (VSDrops)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.20/4812980/sim/;/tests/vsdrops_index.html)
# API diff
:white_check_mark: API Diff from stable
<details><summary>View API diff</summary>

* iOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.20/4812980/apidiff/;/ios-api-diff.html) [gist](https://gist.github.com/aee5ef936b6c399fb4e20e8eebec08f8)
* macOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.20/4812980/apidiff/;/mac-api-diff.html) [gist](https://gist.github.com/7d3332edd83e272ec33ef6c3bb2d77b8)
* macCat [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.20/4812980/apidiff/;/maccat-api-diff.html) [gist](https://gist.github.com/46604c65e1ea3fb8c8acd07090f072bb)
* macCatiOS  [gist](https://gist.github.com/42c0cb7178e5aaa0dff7655291e4af19)
* tvOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.20/4812980/apidiff/;/tvos-api-diff.html) [gist](https://gist.github.com/c60eeb0143af282f2ba64cf09107d825)
* watchOS [vsdrops](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.20/4812980/apidiff/;/watchos-api-diff.html) [gist](https://gist.github.com/139c872d733cc84f6de49cdf059041ce)
</details>

# API & Generator diff

:white_check_mark: [API Diff (from PR only)](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.20/4812980/apigeneratordiff/;//api-diff.html) (no change)
:white_check_mark: [Generator Diff](https://vsdrop.corp.microsoft.com/file/v1/xamarin-macios/device-tests/20210528.20/4812980/apigeneratordiff/;//generator-diff/index.html) (no change)

# GitHub pages

Results can be found in the following github pages (it might take some time to publish):

* [Test results](https://xamarin.github.io/macios.ci/pr/PR11733/4812980/HtmlReport-sim/tests/vsdrops_index.html)
* [API diff ](https://xamarin.github.io/macios.ci/pr/PR11733/4812980/HtmlReport-sim/api-diff/api-diff.html)
* [API & Generator diff](https://xamarin.github.io/macios.ci/pr/PR11733/4812980/apicomparison/api-diff.html)


# Test results

<details>
<summary>2 tests failed, 107 tests passed.</summary>

## Failed tests

 * monotouch-test/Mac [dotnet]/Debug (CoreCLR) [dotnet]: Failed (Test run failed.
Tests run: 2444 Passed: 2354 Inconclusive: 7 Failed: 3 Ignored: 87)
 * monotouch-test/Mac [dotnet]/Debug (CoreCLR, static registrar) [dotnet]: Failed (Test run failed.
Tests run: 2441 Passed: 2354 Inconclusive: 7 Failed: 1 Ignored: 86)
</details>


[Pipeline](https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=4812980&view=ms.vss-test-web.test-result-details) on Agent XAMBOT-1096.BigSur'
Merge 8602b28ea6f49aee557df1d61ca7c1c42c136ad0 into 51c936457d6ce0f7f596179f0fbe6b294c56f6c7

rolfbjarne commented:
Test failures are unrelated:

* monotouch-test/Mac [dotnet]/Debug (CoreCLR) [dotnet]: Fixed in #11732 and #11734.
* monotouch-test/Mac [dotnet]/Debug (CoreCLR, static registrar) [dotnet]: Fixed in #11732.
